subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/WatchableHierarchies.java

```diff
@@ -124,6 +120,20 @@ private SnapshotHierarchy removeUnwatchableFileSystems(SnapshotHierarchy root, I
         return invalidatedRoot;
     }
 
+    private SnapshotHierarchy invalidateUnsupportedFileSystems(SnapshotHierarchy root, Invalidator invalidator) {
+        try {
+            return watchableFileSystemDetector.detectUnsupportedFileSystems()
+                .reduce(
+                    root,
+                    (updatedRoot, fileSystem) -> invalidator.invalidate(fileSystem.getMountPoint().getAbsolutePath(), updatedRoot),
+                    nonCombining()
+                );
+        } catch (NativeException e) {
+            LOGGER.warn("Unable to list file systems to check whether they can be watched. The whole state of the virtual file system has been discarded.", e);
```

Comment:
This is going to be a rather large warning, with a stack trace, right? Or will it only appear if `--stacktrace` is specified? I don't think that's the case. Is this what we want or do we need some more elaborate way to log the exception so that it only appears for `DEBUG` or `INFO`?

Reply:
I honestly don't know if our logging respects the `--stacktrace` flag or not.

We didn't see a case where we couldn't list the file systems, yet. Though the stacktrace probably doesn't help here, much. How about I only add the message from the `NativeException` here to the log message?

Reply:
I thought about this more, and I think `--stacktrace` only affects the output when it's about an exception that bubbled up to the surface and made the build fail. For the exception logged here, we'd always display the stacktrace. And I agree that we shouldn't.

I agree that the actual stacktrace is not very useful here, as we know where the problem comes from. So logging the messages only seems like a good solution.